### PR TITLE
AKS - set up a keyvault and storage account

### PR DESCRIPTION
**Pull request by peteryates**
### Context

This PR contains the first steps towards being able to deploy to AKS.

* [x] set up an [Azure Keyvault](https://azure.microsoft.com/en-gb/products/key-vault)
* [x] set up an [Azure storage account](https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+.PHONY: dev_aks
+dev_aks: aks
+	$(eval include global_config/dev_aks.sh)
+
+read-keyvault-config:
+	$(if $(KEY_VAULT_SECRET_NAME), , $(error Missing environment variable "KEY_VAULT_SECRET_NAME"))
+	$(eval KEY_VAULT_NAME=$(AZURE_RESOURCE_PREFIX)-$(SERVICE_SHORT)-$(CONFIG_SHORT)-kv)
+
+read-deployment-config:
+	$(if $(PLATFORM), , $(error Missing environment variable "PLATFORM"))
```
Comment (saliceti) [inline Makefile:26]: Do we need the PLATFORM variable? In ECF it will always be aks

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+.PHONY: dev_aks
+dev_aks: aks
+	$(eval include global_config/dev_aks.sh)
+
+read-keyvault-config:
+	$(if $(KEY_VAULT_SECRET_NAME), , $(error Missing environment variable "KEY_VAULT_SECRET_NAME"))
+	$(eval KEY_VAULT_NAME=$(AZURE_RESOURCE_PREFIX)-$(SERVICE_SHORT)-$(CONFIG_SHORT)-kv)
+
+read-deployment-config:
```
Comment (saliceti) [inline Makefile:25]: Do we need read-deployment-config at all?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+SHELL						:=/bin/bash
+
+SERVICE_SHORT=cpdecf
+
+.PHONY: help # default target
+help: ## Show this help
+	@grep -E '^[a-zA-Z\.\-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+.PHONY: aks
+aks:
```
Comment (saliceti) [inline Makefile:11]: These 2 variables can be hardcoded instead

File: global_config/development_aks.sh
Comment (peteryates) [inline global_config/development_aks.sh:1]: renamed from `dev` to `development` for consistency, suggest we stick to one throughout

**Review by saliceti — APPROVED** [review]

File: terraform/aks/development_ask.tfvars [hunk 4]
```diff
@@ -0,0 +1,4 @@
+resource_group_name  = "s189t01-afqts-dv-rg"
+storage_account_name = "s189t01afqtstfstatedvsa"
+container_name       = "afqts-tfstate"
```
Comment (smneal) [inline terraform/aks/development_ask.tfvars:3]: We should change `afqts` to `cpdecf`.

Comment (peteryates) [reply terraform/aks/development_ask.tfvars:3]: Eek, will fix that right after lunch

**Comment by saliceti** [comment]
[ECF: Allow review apps to be deployed to AKS](https://trello.com/c/mmO6LoNM/313-ecf-allow-review-apps-to-be-deployed-to-aks)

**Pull request merged**